pandas/core/frame.py

```diff
@@ -1369,7 +1385,10 @@ def from_dict(cls, data, orient="columns", dtype=None, columns=None) -> DataFram
         return cls(data, index=index, columns=columns, dtype=dtype)
 
     def to_numpy(
-        self, dtype=None, copy: bool = False, na_value=lib.no_default
+        self,
+        dtype: Optional[Dtype] = None,
```

Comment:
`dtype` is passed onto `np.array` constructor. `Dtype` includes `ExtensionDtype`, which I think would probably raise at runtime and create mypy errors when we add numpy types shortly.

maybe best to leave untyped for now pending being able to use a dtype alias from numpy shortly.

Reply:
I see we recently added `NpDtype` to `pd._typing`. Shall I use that here?

Reply:
I have reservations on using NpDtype (and adding types to dtype parameters in general, such as #38808) as while these types resolve to `Any`, mypy will not report any inconsistencies.

The issue here will be picked up when we have the numpy types and therefore the burden on reviewers is reduced.

My reservations on adding these ahead of time is that there will likely be more errors to resolve when we transition to using numpy types. Although that boat has already sailed, xref #36092

so rather than doing anything more here now, I'm happy to accept that mypy is green.

Reply:
Ok. Removed `NpDtype` usage here (much easier to revisit after #36092, where there seems to be a lot to do already, has gone in)